core/src/main/java/org/vertexium/GraphBaseWithSearchIndex.java

```diff
@@ -181,30 +182,28 @@ public SearchIndexSecurityGranularity getSearchIndexSecurityGranularity() {
             IndexHint indexHint = m.getIndexHint();
             m.setIndexHint(IndexHint.DO_NOT_INDEX);
             Element element = m.save(authorizations);
+            m.setIndexHint(indexHint);
             elements.add(element);
             if (indexHint == IndexHint.INDEX) {
                 elementsToAddToIndex.add(element);
-                //noinspection unchecked
-                extendedDataToIndex.add(new ElementAndIterableExtendedDataMutation(element, m.getExtendedData()));
             }
         }
         getSearchIndex().addElements(this, elementsToAddToIndex, authorizations);
-        for (ElementAndIterableExtendedDataMutation ed : extendedDataToIndex) {
-            getSearchIndex().addElementExtendedData(this, ed.element, ed.extendedData, authorizations);
+        for (ElementMutation<? extends Element> m : mutations) {
+            if (m.getIndexHint() == IndexHint.INDEX) {
```

Comment:
If you flip the order of the `for` and the `if` you can save a little work by not having to iterate.

Reply:
Looks like the mutation itself has the fetch hints so can't switch this around

Reply:
Yep. I missed that part.